frontend/views/containers/user-settings/NotificationSettings.vue

```diff
@@ -1,20 +1,52 @@
 <template lang='pug'>
   .settings-container
-    .card
-      i18n.link(
-        tag='button'
-        @click.prevent='requestPermission'
-      ) Request Permission
+    section.card
+      i18n.is-title-2.c-title(tag='h2') Settings
+      .c-subcontent
+        .c-text-content
+          i18n.c-smaller-title(tag='h3') Notification Permission
+          i18n.help(tag='p') When you get chat mentions, browser notification will pop up.
+        label
+          i18n.sr-only Notification Permission
+          input.switch(
+            ref='permission'
+            type='checkbox'
+            name='switch'
+            :checked='granted'
+            @change='handleNotificationSettings'
+          )
 </template>
 
 <script>
-import { requestNotificationPermission } from '@model/contracts/shared/nativeNotification.js'
+import { L } from '@common/common.js'
+import {
+  requestNotificationPermission,
+  makeNotification
+} from '@model/contracts/shared/nativeNotification.js'
 
 export default ({
   name: 'NotificationSettings',
+  data () {
+    return {
+      granted: false
+    }
+  },
+  mounted () {
+    this.granted = Notification?.permission === 'granted'
+  },
   methods: {
-    requestPermission () {
-      requestNotificationPermission(true)
+    async handleNotificationSettings (e) {
+      let permission = Notification?.permission
+      if (permission !== null) {
+        permission = await requestNotificationPermission(true)
+      }
+      if (!this.granted && (permission === 'granted')) {
+        makeNotification({
+          title: L('Congratulations'),
+          body: L('You have granted browser notification!')
+        })
+      }
+      this.$refs.permission.checked = this.granted = permission === 'granted'
```

Comment:
> There's no way to tell Firefox to disable the notifications once the permissions have been granted.

Yeah. It's same in the Google Chrome. So what I made this.
```
this.$refs.permission.checked = this.granted = permission === 'granted'
```
So switch will not be changed unless user reset the notification permission and set the permission again.

Reply:
It's same as if user wants to enable after he denied the permission.
User needs to reset permission anyway.

Reply:
@Silver-IT If we are offering the user a checkbox, then the user should be able to easily disable the notifications using this checkbox. Right now it doesn't work - clicking on the checkbox does nothing after notifications permissions have been granted.

We need this to behave in the way users expect: for notifications to stop when they uncheck the checkbox. And since there is no API to disable notifications permissions, we do have to add an extra variable to the state.

Reply:
Yeah. This is what I thought.
But users will see that switch component doesn't work unless they reset notification permission.
Could it be helpful for the users to know why it doesn't work and make him to reset browser permission?

Reply:
Yep! Per the discussion on Slack, in such instances we can let them know via an `alert`